app/workers/worker_base.rb

```diff
@@ -1,7 +1,30 @@
 class WorkerBase
   include Sidekiq::Worker
 
+  def self.perform_async(*args)
+    args << request_id_argument
+    super(*args)
+  end
+
   def self.perform_async_in_queue(queue, *args)
+    args << request_id_argument
     client_push('class' => self, 'args' => args, 'queue' => queue || get_sidekiq_options["queue"])
   end
+
+  def self.request_id_argument
+    {request_id: GdsApi::GovukHeaders.headers[:govuk_request_id]}
+  end
+  private_class_method :request_id_argument
+
+  def perform(*args)
+    last_arg = args.last
+
+    if last_arg.is_a?(Hash) && last_arg.keys == ["request_id"]
+      args.pop
+      request_id = last_arg["request_id"]
```

Comment:
Ah, true, but then if `request_id` is specified as part of other options, then the header won't be set. I think this needs to be:

```ruby
def perform(*args)
  last_arg = args.last

  if last_arg.is_a?(Hash) && last_arg.keys.include? "request_id"
    request_id = last_arg.delete("request_id")
    GdsApi::GovukHeaders.set_header(:govuk_request_id, request_id)
    args.pop if last_arg.empty?
  end

  call(*args)
end
```